### ISSUE TITLE
Microtuning

### ISSUE THREAD

**Issue by suhr** [kometbomb/klystrack#247]

**Comment by kometbomb** [comment]
Care to tell me more? :) You can already do some rudimentary microtuning
with the instrument base note parameter and the slide commands.

to 20. syysk. 2018 klo 10.12 Сухарик (notifications@github.com) kirjoitti:

**Comment by suhr** [comment]
Well, there's some [blog post](http://sevish.com/2016/microtonal-music-trackers/) with examples.

Traditional note names are somewhat problematic, but not a big deal: you can always use several instances of the same instrument (though there is [a more flexible way](https://github.com/suhr/med#note-syntax) to name notes). What matters is the ability to specify the tuning of each note once, without using pitch bend for every note.

**Comment by kometbomb** [comment]
I think this could be achieved with a new command that allows you to slide
just on the very first tick of a row and it takes a value like 00-FF that
would be the percentage part of the microtunal note (another command would
be slide down, just like 01xx and 02xx). Slides work on note values in
klystrack so you could in theory slide to exactly the halfway of two
semitones with such a command.

to 20. syysk. 2018 klo 21.04 Сухарик (notifications@github.com) kirjoitti:

**Comment by suhr** [comment]
The problem is, you have to slide each note. Microtonal tunings don't only divide the semitone in half, they also change the intonation of ordinary notes. For example, in 31-tet you have triads that are close to [pure triads](https://www.youtube.com/watch?v=QzVN1FEhYpU).

Setting the tuning for each note all the time would be very tedious. Better to have something like “tuning programs” in the [MIDI Tuning Standard](http://www.microtonal-synthesis.com/MIDItuning.html).

**Comment by kometbomb** [comment]
Yeah, I get it. Basically the file format allows for 256 different notes
and nothing stops those from being freely tuned to any frequency in any
order. The minimal feasible feature could be to include a frequency to note
map in the song files and user definable tunings (just like the key
mappings).

pe 21. syysk. 2018 klo 15.11 Сухарик (notifications@github.com) kirjoitti:

**Comment by dotmatrixgirl** [comment]
I'd like to bump this, support for scala files would be really nice
